### PR TITLE
Update Xterm and add some X app deps and utils

### PR DESCRIPTION
**Pull request by satmandu**
Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=x2 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/ncompress.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  license 'PublicDomain'
+  compatibility 'all'
+  source_url 'https://github.com/vapier/ncompress/archive/v5.0.tar.gz'
+  source_sha256 '96ec931d06ab827fccad377839bfb91955274568392ddecf809e443443aead46'
+
+  binary_url({
+     aarch64: 'file:///usr/local/tmp/packages/ncompress-5.0-chromeos-armv7l.tar.zst',
+      armv7l: 'file:///usr/local/tmp/packages/ncompress-5.0-chromeos-armv7l.tar.zst',
+        i686: 'file:///usr/local/tmp/packages/ncompress-5.0-chromeos-i686.tar.zst',
+      x86_64: 'file:///usr/local/tmp/packages/ncompress-5.0-chromeos-x86_64.tar.zst'
```
Comment (uberhacker) [inline packages/ncompress.rb:19]: Local files?

File: packages/xterm.rb [hunk 2]
```diff
@@ -3,46 +3,83 @@
...
+      --enable-wide-chars \
+      --with-app-defaults=#{CREW_PREFIX}/share/X11/app-defaults/ \
+      --with-pcre \
+      --with-pkg-config \
+      --with-tty-group=tty \
+      --with-utempter \
+      --with-x \
+      --with-xinerama \
+      --with-xpm=#{CREW_PREFIX}"
+    system "sed -i '/^LDFLAGS/ s/$/ -DUSE_TERMINFO=1 -ltinfo/' Makefile"
```
Comment (uberhacker) [inline packages/xterm.rb:82]: Move to `def self.patch` section?

Comment (satmandu) [reply packages/xterm.rb:82]: Can't, as configure needs to run first.

But I'm having a discussion upstream about how to do this cleanly...

File: packages/ncompress.rb (hunk 1 above)
Comment (satmandu) [reply packages/ncompress.rb]: Fixed!

File: packages/xterm.rb (hunk 2 above)
Comment (uberhacker) [reply packages/xterm.rb:82]: Can it be patched in `Makefile.in` prior to the `configure`?

Comment (satmandu) [reply packages/xterm.rb:82]: Possibly... but I'm waiting for an upstream fix to resolve the issue without needing patching...

I've sent my config.status and config.log to the maintainer as requested...

Comment (satmandu) [reply packages/xterm.rb]: I just went ahead and added a patch section to modify the Makefile.in...

**Review by uberhacker — APPROVED** [review]

**Pull request merged**